### PR TITLE
use msys2 on Windows

### PR DESCRIPTION
**Pull request by marten-seemann**
This is how our workflows would look like once we merge https://github.com/protocol/multiple-go-modules/pull/8.

### CODE REVIEW

**Comment by marten-seemann** [comment]
I updated the PR to use [v1.3 of multiple-go-modules](https://github.com/protocol/multiple-go-modules/releases/tag/v1.3).

File: templates/.github/workflows/go-test.yml [hunk 1]
```diff
@@ -6,12 +6,15 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ "ubuntu", "windows", "macos" ]
         go: [ "1.16.x", "1.17.x" ]
+        os:
+          - { name: "ubuntu", shell: "bash" }
+          - { name: "macos", shell: "bash" }
+          - { name: "windows", shell: "msys2 {0}" }
```
Comment (mvdan) [inline templates/.github/workflows/go-test.yml:13]: I wonder if we could derive the shell by default from the OS name, to reduce verbosity.

**Review by mvdan — APPROVED** [review]

File: templates/.github/workflows/go-test.yml (hunk 1 above)
Comment (marten-seemann) [reply templates/.github/workflows/go-test.yml:13]: Not without a separate step I think.

Comment (galargh) [reply templates/.github/workflows/go-test.yml:13]: NOTE: this is a breaking change if someone started using `os.name` in their custom `go-test-setup/action.yml`. I don't expect there are many (if any) places like this so I can just keep an eye out for failing sync PRs once we propagate this change throughout the repos and fix where needed.

**Comment by galargh** [comment]
Closing this since the alternative https://github.com/protocol/.github/pull/232 was merged into `next` branch.

**Pull request closed without merging**